cache_test.go

```diff
@@ -328,6 +310,78 @@ func TestCacheErr(t *testing.T) {
 	}
 }
 
+// TestReadCache verifies that ReadCache successfully reads from an existing cache.
+func TestReadCache(t *testing.T) {
+	dir, cleanup := testutil.TempDir(t, "", "")
+	defer cleanup()
+	prefix := filepath.Join(dir, "cached")
+	ctx := context.Background()
+
+	const (
+		N      = 10000
+		Nshard = 10
+	)
+	input := make([]int, N)
+	for i := range input {
+		input[i] = i
+	}
+	slice1 := bigslice.Const(Nshard, input)
+	slice1 = bigslice.Cache(ctx, slice1, prefix)
+	scan1 := runLocal(ctx, t, slice1)
+	defer scan1.Close()
+
+	// We now have a populated cache. Read from it, and make sure we get the
+	// same results.
+	slice2 := bigslice.ReadCache(ctx, slice1, slice1.NumShard(), prefix)
+	scan2 := runLocal(ctx, t, slice2)
+	defer scan2.Close()
+
+	var n int
+	for {
+		var v1, v2 int
+		ok := scan1.Scan(ctx, &v1)
+		if got, want := scan2.Scan(ctx, &v2), ok; got != want {
+			t.Errorf("got %v, want %v", got, want)
+			break
+		}
+		if !ok {
+			break
+		}
+		if v1 != v2 {
```

Comment:
It seems like this test depends on the local executor scanning shards in consistent order, if I'm not misreading. Is that property guaranteed? Either way, it seems pretty subtle, and maybe worth a comment. Or just scan both separately, then sort and compare.